File: services/order-service/src/main/java/com/bookland/order/entity/PaymentMethod.java
package com.bookland.order.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "payment_method")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PaymentMethod {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String providerCode;

    @Builder.Default
    private Boolean isOnline = false;

    @Column(columnDefinition = "TEXT")
    private String description;
}

File: BookLand_FE/src/types/PaymentMethod.ts
import type { Bill } from './Bill';
import type { PaymentTransaction } from './PaymentTransaction';

export interface PaymentMethod {
    id: number;
    name: string;
    providerCode: string;
    isOnline: boolean;
    description?: string;
    bills?: Bill[];
    transactions?: PaymentTransaction[];
}

export interface PaymentMethodRequest {
    name: string;
    providerCode?: string;
    isOnline: boolean;
    description?: string;
}
